src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.ts

```diff
@@ -348,6 +349,55 @@ export class RgwMultisiteDetailsComponent implements OnDestroy, OnInit {
     return this.showMigrateAction;
   }
 
+  isDeletedDisabled(node: TreeNode): boolean {
+    let disable: boolean = false;
+    let nMasterZonegroup: number = 0;
+    if (node.data.type === 'realm' && node.data.is_default && this.realms.length < 2) {
+      disable = true;
+    }
+
+    if (node.data.type === 'zonegroup') {
+      if (this.zonegroups.length < 2) {
+        this.deleteTitle = 'You can not delete the only zonegroup available';
```

Comment:
I supposed this function would only return whether it is disabled or not. Maybe the return should be an empty string when it isn't disabled and a message of why is disabled if true.